drivers/platform/stm32/stm32_spi.c

```diff
@@ -212,9 +216,45 @@ int32_t stm32_spi_init(struct no_os_spi_desc **desc,
 	if (ret)
 		goto error;
 
+	if (sinit->dma_init) {
```

Comment:
everything related to DMA from stm32_spi shoul use #ifdef HAL_DMA_MODULE_ENABLED and everything related to PWM should use HAL_TIM_MODULE_ENABLED